app/scripts/app.js

```diff
@@ -35,6 +35,16 @@ angular
     'openshiftCommonUI',
     'webCatalog'
   ])
+  .run(function(AuthService, HomePagePreference, $location) {
+    if ($location.$$url === "/") {
+      HomePagePreference.gotoHomePage();
+    }
+    AuthService.onUserChanged(function(user) {
+      if (user) {
+        HomePagePreference.gotoHomePage();
```

Comment:
It still seems like this could break direct linking? I think we need to avoid the `run()` block.